singer_sdk/streams/__init__.py

```diff
@@ -3,12 +3,6 @@
 from singer_sdk.streams.core import Stream
 from singer_sdk.streams.graphql import GraphQLStream
 from singer_sdk.streams.rest import RESTStream
-from singer_sdk.streams.sql import SQLConnector, SQLStream
+from singer_sdk.streams.sql import SQLStream
 
-__all__ = [
-    "Stream",
-    "GraphQLStream",
-    "RESTStream",
-    "SQLStream",
-    "SQLConnector",
-]
+__all__ = ["Stream", "GraphQLStream", "RESTStream", "SQLStream"]
```

Comment:
@kgpayne

I think we could go either way on this but are we sure we want to remove the reference of `singer_sdk.streams.SQLConnector`?

I'm totally okay with the decision to do so, but probably we should also remove `singer_sdk.sinks.SQLConnector` here if so: https://github.com/meltano/sdk/blob/kgpayne/issue1090/singer_sdk/sinks/__init__.py

I'd also be okay leaving it as referenceable from both `singer_sdk.streams` and `singer_sdk.sinks`. Small benefit of not breaking existing code with the slight downside of possible confusion of where to best import from.

cc @edgarrmondragon

Reply:
@kgpayne @aaronsteers

TLDR I'm okay with removing `SQLConnector` from the "public API" of the module.

---

`__all__` only affects _star_ imports.

For example, consider the module

```python
# mymod.py
__all__ = ["X", "Y"]

X = 1
Y = 2
Z = 3
```

Then the module's "public API" is only reflected when importing with `*`:

```python
>>> from mymod import *
>>> X, Y, Z
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'Z' is not defined
```

But the `Z` member can still be imported explicitly:

```python
>>> from mymod import X, Y, Z
>>> X, Y, Z
(1, 2, 3)
```

Reply:
> `__all__` only affects _star_ imports.

Agreed, and understood. I believe it may also affect IDE auto-complete for those editors which will use it to hint at members within those modules, yes?

This SO article seems to confirm: 

https://stackoverflow.com/a/35710527/4298208

> Documentation tools
>
> Documentation and code autocompletion tools may (in fact, should) also inspect the `__all__` to determine what names to show as available from a module.

I do think it's worth being careful about out `__all__` declarations, for the purpose of those IDE hints.

Also - to clarify, I believe the 'breaking' part would be the prior references at the `__init__.py` level, specifically `singer_sdk.[streams|sinks].SQLConnector` - but this would still work `singer_sdk.[streams|sinks].sql.SQLConnector` because both `.../sql.py` files import `SQLConnector` at the top of the file.

Reply:
Done ✅